core/src/main/java/com/amplifyframework/auth/AuthUserAttributeKey.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth;
+
+import androidx.annotation.NonNull;
+import androidx.core.util.ObjectsCompat;
+
+public final class AuthUserAttributeKey {
```

Comment:
Hm. There is a lot of "opinion" in this class, in core. How many of these attributes come from Cognito?

Reply:
All of them - however, none of these are specific to the Cognito implementation so I think it's fair to start with this as the initial common list.